src/main/java/graphql/parser/ExtendedBailStrategy.java

```diff
@@ -0,0 +1,89 @@
+package graphql.parser;
+
+import graphql.InvalidSyntaxError;
+import graphql.language.SourceLocation;
+import org.antlr.v4.runtime.BailErrorStrategy;
+import org.antlr.v4.runtime.Parser;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.misc.ParseCancellationException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+
+public class ExtendedBailStrategy extends BailErrorStrategy {
+    private final String input;
+    private final String sourceName;
+
+    public ExtendedBailStrategy(String input, String sourceName) {
+        this.input = input;
+        this.sourceName = sourceName;
+    }
+
+    @Override
+    public void recover(Parser recognizer, RecognitionException e) {
+        try {
+            super.recover(recognizer, e);
+        } catch (ParseCancellationException parseException) {
+            throw mkException(recognizer, e);
+        }
+    }
+
+    @Override
+    public Token recoverInline(Parser recognizer) throws RecognitionException {
+        try {
+            return super.recoverInline(recognizer);
+        } catch (ParseCancellationException parseException) {
+            throw mkException(recognizer, null);
+        }
+    }
+
+    private InvalidSyntaxError mkException(Parser recognizer, RecognitionException cause) {
+        int line = -1;
+        int column = -1;
+        String msg = null;
+        String sourcePreview = null;
+        Token currentToken = recognizer.getCurrentToken();
+        if (currentToken != null) {
+            line = currentToken.getLine();
+            column = currentToken.getCharPositionInLine();
+            sourcePreview = mkPreview(line);
+            msg = "Offending token is '" + currentToken.getText() + "'";
+        }
+        return new InvalidSyntaxError(singletonList(new SourceLocation(line, column, sourceName)), msg, sourcePreview, cause);
```

Comment:
can the resulting message end up having a `-1,-1` location if `currentToken` result in a `null` value?

Reply:
fixed